controllers/component_controller_conditions.go

```diff
@@ -99,18 +100,18 @@ func (r *ComponentReconciler) SetUpdateConditionAndUpdateCR(ctx context.Context,
 	err := r.Client.Status().Update(ctx, component)
 	if err != nil {
 		// Retry, and if still fails, then return an error
-		curComponent := &appstudiov1alpha1.Component{}
-		err := r.Client.Get(ctx, req.NamespacedName, curComponent)
+		var currentComponent appstudiov1alpha1.Component
+		err := r.Get(ctx, req.NamespacedName, &currentComponent)
 		if err != nil {
-			return err
+			return nil
```

Comment:
Same comment as before

Reply:
updated to return error if fail to get component at the retry